XsltPipe 'IOException Stream closed' when adapter is called multiple times concurrently and leaves other adapters unable to handle messages

### Current Behavior

Long time issue that we couldn't reproduce for a long time. Seems that it only happens when the adapter is called multiple times concurrently. The exception is thrown by the first XsltPipe `LegalizeSoapActionForPipeName` in the adapter that is only doing a simple xpath `replace()`. 

When this issue takes place, the framework ends up in a state where at least none of the other adapters with WebserviceListeners are able to accept messages anymore, until the framework is completely restarted. 

### Environment Information

**FF! 9.0.0-20241026.042336**: ZaakBrug 1.22.7
Running on **afd24d1ffd3f** using **Apache Tomcat/10.1.28**
Java Version: **OpenJDK Runtime Environment (21.0.4+7-LTS)**
Heap size: **266.9 MiB**, total JVM memory: **864.0 MiB**
Free memory: **597.1 MiB**, max memory: **11.5 GiB**
Free disk space: **989.5 GiB**, total disk space: **1006.9 GiB**

### Steps To Reproduce

Call adapter multiple times concurrently

### Configuration

```xml
<Module xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="../FrankConfig.xsd">
    <Adapter name="SoapEndpointRouter_VrijeBerichten"
        active="${SoapEndpointRouter_VrijeBerichten.Active}"
        description="">

        <!--
            Never use "originalMessage" in this adapter. Use "SafeOriginalMessage" instead. 
            This is due "originalMessage" potentially containing very large document base64. 
            "SafeOriginalMessage" has the base64 already stored to disk and removed from the message.
        -->

        <Receiver name="SoapEndpointRouter_VrijeBerichten">
            <WebServiceListener name="SoapEndpointRouter_VrijeBerichten" address="${zaakbrug.soap.vrije-berichten.endpoint}" soap="false"/>
        </Receiver>

        <Pipeline>
            <Exits>
                <Exit name="EXIT" state="SUCCESS"/>
                <Exit name="EXCEPTION" state="ERROR"/>
            </Exits>

            <!--
                Dont touch the input message before this pipe. Especially no xslt/xpath. 
                It can potentially contain very large document base64 still. 
                This pipe stores it to disk and removes it from the message.
            -->
            <EchoPipe 
                name="StoreInhoud"
                elementToMove="inhoud"
                removeCompactMsgNamespaces="false"
                >
                <Forward name="success" path="CheckInputDocumentObjectCount" />
            </EchoPipe>

            <!--
                In the rare case of the new base64 AND the old base64 of a document being present during an updateDocument action, the elementsToMove will make a
                ref_inhoud# sessionKey for both "inhoud" elements. Normally there is only one because there is only 1 inhoud element with the NEW base64.
                Only in this scenario "ref_inhoud" will be the OLD base64 and "ref_inhoud2" will be the NEW base64. We are only interested in the NEW base64,
                so we are overwriting the "ref_inhoud" sessionkey with the content of "ref_inhoud2" and replace "{sessionKey:ref_inhoud2} with "{sessionKey:ref_inhoud}".
            -->
            <XmlIfPipe name="CheckInputDocumentObjectCount"
                xpathExpression="count(//*:inhoud) = 2"
                preserveInput="true"
                >
                <Forward name="then" path="OverwriteRefInhoudWithRefInhoud2"/>
                <Forward name="else" path="ReplaceInhoudRefWithBase64EncodedRef"/>
            </XmlIfPipe>

            <PutInSessionPipe
                name="OverwriteRefInhoudWithRefInhoud2"
                >
                <Param name="ref_inhoud" sessionKey="ref_inhoud2" />
                <Forward name="success" path="ReplaceInhoudRefWithBase64EncodedRef"/>
                <Forward name="exception" path="BackEndError"/>
            </PutInSessionPipe>

            <ReplacerPipe
                name="ReplaceInhoudRefWithBase64EncodedRef"
                storeResultInSessionKey="SafeOriginalMessage"
                find="{sessionKey:ref_inhoud}"
                replace="e3Nlc3Npb25LZXk6cmVmX2luaG91ZH0="
                >
                <Forward name="success" path="ReplaceInhoudRef2WithBase64EncodedRef" />
            </ReplacerPipe>

            <ReplacerPipe
                name="ReplaceInhoudRef2WithBase64EncodedRef"
                storeResultInSessionKey="SafeOriginalMessage"
                find="{sessionKey:ref_inhoud2}"
                replace="e3Nlc3Npb25LZXk6cmVmX2luaG91ZH0="
                >
                <Forward name="success" path="ValidateInput" />
            </ReplacerPipe>

            <WsdlXmlValidatorPipe
                name="ValidateInput"
				wsdl="Common/xsd/Zaak_DocumentServices_1_1_02/zkn0310/zs-dms/zkn0310_vrijeBerichten_zs-dms.wsdl"
                outputFormat="XML"
                ignoreUnknownNamespaces="true"
                autoFormat="false"
                >
				<Forward name="success" path="UnwrapOriginalMessage"/>
				<Forward name="failure" path="WsdlValidationSoftFailForwarder"/>
			</WsdlXmlValidatorPipe>

            <XmlSwitchPipe
                name="WsdlValidationSoftFailForwarder"
                xpathExpression="$SoftFail = true()"
                >
                <Param name="SoftFail" value="${zaakbrug.soap.vrije-berichten.validation-soft-fail}" type="BOOLEAN"/>
                <Forward name="true" path="UnwrapOriginalMessage" />
                <Forward name="false" path="InvalidXml" />
            </XmlSwitchPipe>

            <SoapWrapperPipe
                name="UnwrapOriginalMessage"
                direction="UNWRAP"
                removeOutputNamespaces="true"
                >
                <Forward name="success" path="StoreResponeContextSessionKeys"/>
                <Forward name="exception" path="BackEndError"/>
            </SoapWrapperPipe>

            <PutInSessionPipe
                name="StoreResponeContextSessionKeys"
                >
                <Param name="Stuurgegevens" xpathExpression="/*/stuurgegevens" defaultValue="&lt;stuurgegevens/&gt;" type="DOMDOC" />
                <Param name="Parameters" xpathExpression="/*/parameters" defaultValue="&lt;parameters/&gt;" type="DOMDOC" />
                <Forward name="success" path="LegalizeSoapActionForPipeName"/>
                <Forward name="exception" path="BackEndError"/>
            </PutInSessionPipe>

            <XsltPipe
                name="LegalizeSoapActionForPipeName"
                preserveInput="true"
                xpathExpression="replace($Action, '/', '_')"
                storeResultInSessionKey="LegalizedSoapAction"
                >
                <Param name="Action" sessionKey="SOAPAction"/>
                <Forward name="success" path="checkObjectCount"/>
                <Forward name="exception" path="BackEndError"/>
            </XsltPipe>

            <XmlIfPipe name="checkObjectCount" xpathExpression="count(/*/edcLk02/object)=2">
                <Forward name="then" path="StoreZdsWordtInformatieObject1" />
                <Forward name="else" path="StoreZdsWordtInformatieObject2" />
            </XmlIfPipe>

            <PutInSessionPipe name="StoreZdsWordtInformatieObject1">
                <Param name="ZdsWordtInformatieObject" xpathExpression="/*/edcLk02/object[2]" defaultValue="&lt;object/&gt;" type="DOMDOC" />
                <Forward name="success" path="ForwardBySoapActionSwitch" />
            </PutInSessionPipe>

            <PutInSessionPipe name="StoreZdsWordtInformatieObject2">
                <Param name="ZdsWordtInformatieObject" xpathExpression="/*/edcLk02/object[1]" defaultValue="&lt;object/&gt;" type="DOMDOC" />
                <Forward name="success" path="ForwardBySoapActionSwitch" />
            </PutInSessionPipe>

            <XmlSwitchPipe
                name="ForwardBySoapActionSwitch"
				forwardNameSessionKey="LegalizedSoapAction"
                notFoundForwardName="UnsupportedSoapActionError"
                />
            
            <!-- GenereerZaakIdentificatie_Di02 -->
            <SenderPipe
                name="http:__www.egem.nl_StUF_sector_zkn_0310_genereerZaakIdentificatie_Di02"
                storeResultInSessionKey="Identificatie"
                >
                <IbisLocalSender
                    name="GenereerZaakIdentificatie_Di02LocalSender"
                    javaListener="GenereerIdentificatieEmulator"
                    returnedSessionKeys="Error">
                    <Param name="IdentificatieType" value="ZaakIdentificatie"/>
                </IbisLocalSender>
                <Forward name="success" path="WrapGenereerZaakIdentificatie_Du02Response" />
                <Forward name="exception" path="BackEndError" />
            </SenderPipe>

            <!-- GenereerDocumentIdentificatie_Di02 -->
            <SenderPipe
                name="http:__www.egem.nl_StUF_sector_zkn_0310_genereerDocumentIdentificatie_Di02"
                storeResultInSessionKey="Identificatie"
                >
                <IbisLocalSender
                    name="GenereerDocumentIdentificatie_Di02LocalSender"
                    javaListener="GenereerIdentificatieEmulator"
                    returnedSessionKeys="Error">
                    <Param name="IdentificatieType" value="DocumentIdentificatie"/>
                </IbisLocalSender>
                <Forward name="success" path="WrapGenereerDocumentIdentificatie_Di02Response" />
                <Forward name="exception" path="BackEndError" />
            </SenderPipe>

            <!-- GeefZaakdocumentbewerken_Di02 -->
            <SenderPipe
                name="http:__www.egem.nl_StUF_sector_zkn_0310_geefZaakdocumentbewerken_Di02"
                storeResultInSessionKey="ZdsZaakDocumentInhoud"
                >
                <IbisLocalSender
                    name="GeefZaakdocumentbewerken_Di02LocalSender"
                    returnedSessionKeys="ZgwLock,ref_inhoud,Error"
                    javaListener="GeefZaakdocumentbewerken_Di02"
                    >
                    <Param name="DocumentIdentificatie" xpathExpression="/*/edcLv01/gelijk/identificatie" type="STRING" />
                </IbisLocalSender>
                <Forward name="success" path="WrapGeefZaakdocumentbewerken_Di02Response"/>
                <Forward name="exception" path="EXCEPTION" />
            </SenderPipe>

            <!-- CancelCheckout_Di02 -->
            <SenderPipe
                name="http:__www.egem.nl_StUF_sector_zkn_0310_cancelCheckout_Di02"
                >
                <IbisLocalSender
                    name="CancelCheckout_Di02LocalSender"
                    javaListener="CancelCheckOutZaakDocument_Di02"
                    returnedSessionKeys="Error">
                    <Param name="DocumentIdentificatie" xpathExpression="/*/document/identificatie" type="STRING" />
                    <Param name="Lock" xpathExpression="/*/parameters/checkedOutId" type="STRING" />
                </IbisLocalSender>
                <Forward name="success" path="WrapCheckOutZaakDocument_Di02Response"/>
                <Forward name="exception" path="EXCEPTION" />
            </SenderPipe>

            <!-- UpdateZaakdocument_Di02 -->
            <SenderPipe
                name="http:__www.egem.nl_StUF_sector_zkn_0310_updateZaakdocument_Di02"
                >
                <IbisLocalSender
                    name="UpdateZaakdocument_Di02LocalSender"
                    javaListener="UpdateZaakDocument_Di02"
                    returnedSessionKeys="Error">
                    <Param name="ZdsWordtInformatieObject" sessionKey="ZdsWordtInformatieObject" />
                    <Param name="Lock" xpathExpression="/*/parameters/checkedOutId" type="STRING" />
                    <Param name="ref_inhoud" sessionKey="ref_inhoud" />
                </IbisLocalSender>
                <Forward name="success" path="WrapUpdateZaakDocument_Di02Response"/>
                <Forward name="exception" path="EXCEPTION" />
            </SenderPipe>

            <SoapWrapperPipe
                name="WrapGenereerZaakIdentificatie_Du02Response"
                getInputFromFixedValue="&lt;EmptyInput/&gt;"
                soapBodyStyleSheet="Common/xsl/CreateGenereerZaakIdentificatie_Du02Response.xslt"
                >
                <Param name="Stuurgegevens" sessionKey="Stuurgegevens" type="DOMDOC" />
                <Param name="Identificatie" sessionKey="Identificatie" type="STRING" />
                <Forward name="success" path="EXIT"/>
            </SoapWrapperPipe>

            <SoapWrapperPipe
                name="WrapGenereerDocumentIdentificatie_Di02Response"
                getInputFromFixedValue="&lt;EmptyInput/&gt;"
                soapBodyStyleSheet="Common/xsl/CreateGenereerDocumentIdentificatie_Du02Response.xslt"
                >
                <Param name="Stuurgegevens" sessionKey="Stuurgegevens" type="DOMDOC" />
                <Param name="Identificatie" sessionKey="Identificatie" type="STRING" />
                <Forward name="success" path="EXIT"/>
            </SoapWrapperPipe>

            <SoapWrapperPipe
                name="WrapGeefZaakdocumentbewerken_Di02Response"
                getInputFromFixedValue="&lt;EmptyInput/&gt;"
                soapBodyStyleSheet="Common/xsl/CreateDu02Response.xslt"
                >
                <Param name="Stuurgegevens" sessionKey="Stuurgegevens" type="DOMDOC" />
                <!-- <Param name="Parameters" sessionKey="Parameters" type="DOMDOC"/> -->
                <Param name="ZgwLock" sessionKey="ZgwLock" type="DOMDOC" />
                <Param name="ZdsZaakDocumentInhoud" sessionKey="ZdsZaakDocumentInhoud" type="DOMDOC" />
                <Forward name="success" path="ReplaceBase64EncodedRefWithInhoudRef"/>
            </SoapWrapperPipe>

            <ReplacerPipe
                name="ReplaceBase64EncodedRefWithInhoudRef"
                find="e3Nlc3Npb25LZXk6cmVmX2luaG91ZH0="
                replace="?{ref_inhoud}"
                >
                <Forward name="success" path="SubstituteDocumentDataRef" />
            </ReplacerPipe>

            <ReplacerPipe
                name="SubstituteDocumentDataRef"
                >
                <Param name="ref_inhoud" sessionKey="ref_inhoud" />
                <Forward name="success" path="EXIT" />
            </ReplacerPipe> 

            <SoapWrapperPipe
                name="WrapCheckOutZaakDocument_Di02Response"
                getInputFromFixedValue="&lt;EmptyInput/&gt;"
                soapBodyStyleSheet="Common/xsl/CreateBv02Response.xslt"
                >
                <Param name="Stuurgegevens" sessionKey="Stuurgegevens" type="DOMDOC" />
                <Forward name="success" path="EXIT"/>
            </SoapWrapperPipe>

            <SoapWrapperPipe
                name="WrapUpdateZaakDocument_Di02Response"
                getInputFromFixedValue="&lt;EmptyInput/&gt;"
                soapBodyStyleSheet="Common/xsl/CreateBv02Response.xslt"
                >
                <Param name="Stuurgegevens" sessionKey="Stuurgegevens" type="DOMDOC" />
                <Forward name="success" path="EXIT"/>
            </SoapWrapperPipe>

            <!-- ERRORS -->
            <XsltPipe 
                name="UnsupportedSoapActionError"
                getInputFromFixedValue="&lt;dummy/&gt;"
                styleSheetName="Common/xsl/BuildError.xsl"
                storeResultInSessionKey="Error">
                <Param name="cause" sessionKey="Error" type="DOMDOC" />
                <Param name="code" value="TechnicalError" /> <!-- codes: TechnicalError, TranslationError, ConfigurationError-->
                <Param name="reason" pattern="Unsupported SOAP-action. No implementation found for: {SOAPAction}" />
                <!-- <Param name="details" sessionKey="" /> -->
                <!-- <Param name="detailsXml" type="DOMDOC" /> -->
                <Forward name="success" path="BackEndError" />
                <Forward name="exception" path="BackEndError" />
            </XsltPipe>

            <PutInSessionPipe name="InvalidXml">
                <Param name="errorCode" value="INVALID_XML"/>
                <Param name="errorReason" value="XML was invalid"/>
                <Forward name="success" path="WrapFo03Response" />
            </PutInSessionPipe>

            <PutInSessionPipe name="NotWellFormed">
                <Param name="errorCode" value="NOT_WELL_FORMED_XML"/>
                <Param name="errorReason" value="XML was not according to xsd"/>             
                <Forward name="success" path="WrapFo03Response"/>
            </PutInSessionPipe>

            <PutInSessionPipe
                name="BackEndError"
                unlessSessionKey="errorCode"
                >
                <Param name="errorCode" styleSheetName="Common/xsl/BackEndError.xsl" />
                <Param name="errorReason" xpathExpression="(//error)[last()]/reason" />
                <Param name="errorDetailText" xpathExpression="(//error)[last()]/details" />
                <Param name="errorDetailXML" xpathExpression="/error" type="DOMDOC" />
                <Forward name="success" path="WrapFo03Response" />
            </PutInSessionPipe>

            <SoapWrapperPipe
                name="WrapFo03Response"
                emptyInputReplacement="&lt;EmptyInput/&gt;"
                soapBodyStyleSheet="Common/xsl/CreateFo03Response.xslt"
                >
                <Param name="Stuurgegevens" sessionKey="Stuurgegevens" type="DOMDOC" />
                <Param name="Parameters" sessionKey="Parameters" type="DOMDOC" />
                <Param name="errorCode"   sessionKey="errorCode" />
                <Param name="errorReason" sessionKey="errorReason" />
                <Param name="errorDetailText" sessionKey="errorDetailText" />
                <Param name="errorDetailXML" sessionKey="errorDetailXML" type="DOMDOC" />
                <Forward name="success" path="EXIT"/>
            </SoapWrapperPipe>
        </Pipeline>
    </Adapter>
</Module>
```


### Input

```xml
<SOAP-ENV:Envelope xsi:schemaLocation="http://www.egem.nl/StUF/sector/zkn/0310 zkn0310-p28\zs-dms\zkn0310_msg_zs-dms_resolved.xsd" xmlns:SOAP-ENV="http://schemas.xmlsoap.org/soap/envelope/" xmlns:StUF="http://www.egem.nl/StUF/StUF0301" xmlns:ZKN="http://www.egem.nl/StUF/sector/zkn/0310" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
   <SOAP-ENV:Header/>
   <SOAP-ENV:Body>
      <ZKN:genereerZaakIdentificatie_Di02>
         <ZKN:stuurgegevens>
            <StUF:berichtcode>Di02</StUF:berichtcode>
            <StUF:zender>
               <StUF:organisatie>1900</StUF:organisatie>
               <StUF:applicatie>GWS4all</StUF:applicatie>
               <StUF:gebruiker>Gebruiker</StUF:gebruiker>
            </StUF:zender>
            <StUF:ontvanger>
               <StUF:organisatie>1900</StUF:organisatie>
               <StUF:applicatie>CDR</StUF:applicatie>
            </StUF:ontvanger>
            <StUF:referentienummer>ac150003--773cd1c0_1944a3edfa2_-7fc3</StUF:referentienummer>
            <StUF:tijdstipBericht>20250109104400</StUF:tijdstipBericht>
            <StUF:functie>genereerZaakidentificatie</StUF:functie>
         </ZKN:stuurgegevens>
      </ZKN:genereerZaakIdentificatie_Di02>
   </SOAP-ENV:Body>
</SOAP-ENV:Envelope>
```

### What database are you using?

PostgreSQL

### What browsers are you seeing the problem on?

_No response_

### Relevant Log Output

```console
2025-01-09 09:47:41,819 INFO  [http-nio-8080-exec-5] {mid=ac150003--773cd1c0_1944a3edfa2_-7fd1, cid=ac150003--773cd1c0_1944a3edfa2_-7fd1, adapter=SoapEndpointRouter_VrijeBerichten} core.Adapter - Adapter [SoapEndpointRouter_VrijeBerichten] Pipeline finished processing message with messageId [ac150003--773cd1c0_1944a3edfa2_-7fd1] with exit-state [SUCCESS]
2025-01-09 09:47:41,866 INFO  [http-nio-8080-exec-10] {mid=ac150003--773cd1c0_1944a3edfa2_-7fd0, cid=ac150003--773cd1c0_1944a3edfa2_-7fd0, adapter=SoapEndpointRouter_VrijeBerichten} receivers.Receiver - Receiver [SoapEndpointRouter_VrijeBerichten]  messageId [ac150003--773cd1c0_1944a3edfa2_-7fd0] correlationId [ac150003--773cd1c0_1944a3edfa2_-7fd0] businessCorrelationId [ac150003--773cd1c0_1944a3edfa2_-7fd0]
2025-01-09 09:47:41,867 INFO  [http-nio-8080-exec-10] {mid=ac150003--773cd1c0_1944a3edfa2_-7fd0, cid=ac150003--773cd1c0_1944a3edfa2_-7fd0, adapter=SoapEndpointRouter_VrijeBerichten} core.Adapter - Adapter [SoapEndpointRouter_VrijeBerichten] received message with messageId [ac150003--773cd1c0_1944a3edfa2_-7fd0]
2025-01-09 09:47:41,891 WARN  [http-nio-8080-exec-10] {mid=ac150003--773cd1c0_1944a3edfa2_-7fd0, cid=ac150003--773cd1c0_1944a3edfa2_-7fd0, adapter=SoapEndpointRouter_VrijeBerichten, pipe=LegalizeSoapActionForPipeName} util.TransformerErrorListener - Fatal transformation error: I/O error reported by XML parser processing null: Stream closed
2025-01-09 09:47:41,895 INFO  [http-nio-8080-exec-10] {mid=ac150003--773cd1c0_1944a3edfa2_-7fd0, cid=ac150003--773cd1c0_1944a3edfa2_-7fd0, adapter=SoapEndpointRouter_VrijeBerichten, pipe=LegalizeSoapActionForPipeName} processors.ExceptionHandlingPipeProcessor - exception occured, forwarding to exception-forward [exception], exception:

org.frankframework.core.PipeRunException: Pipe [LegalizeSoapActionForPipeName] Exception on transforming input: Cannot transform input: (XPathException) I/O error reported by XML parser processing null: (IOException) Stream closed
	at org.frankframework.pipes.XsltPipe.doPipe(XsltPipe.java:116) ~[frankframework-core-9.0.0-20241026.042336.jar:9.0.0-20241026.042336]
	... suppressed 15 lines
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(Unknown Source) ~[?:?]
	... suppressed 6 lines
	at org.frankframework.ladybug.IbisDebuggerAdvice.debugPipeGetInputFrom(IbisDebuggerAdvice.java:189) ~[frankframework-ladybug-debugger-9.0.0-20241026.042336.jar:9.0.0-20241026.042336]
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(Unknown Source) ~[?:?]
	... suppressed 18 lines
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(Unknown Source) ~[?:?]
	... suppressed 6 lines
	at org.frankframework.ladybug.IbisDebuggerAdvice.debugPipeInputOutputAbort(IbisDebuggerAdvice.java:162) ~[frankframework-ladybug-debugger-9.0.0-20241026.042336.jar:9.0.0-20241026.042336]
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(Unknown Source) ~[?:?]
	... suppressed 15 lines
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(Unknown Source) ~[?:?]
	... suppressed 6 lines
	at org.frankframework.ladybug.IbisDebuggerAdvice.debugPipeLineInputOutputAbort(IbisDebuggerAdvice.java:127) ~[frankframework-ladybug-debugger-9.0.0-20241026.042336.jar:9.0.0-20241026.042336]
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(Unknown Source) ~[?:?]
	... suppressed 16 lines
	at org.frankframework.core.PipeLine.process(PipeLine.java:411) ~[frankframework-core-9.0.0-20241026.042336.jar:9.0.0-20241026.042336]
	at org.frankframework.core.Adapter.processMessageWithExceptions(Adapter.java:610) ~[frankframework-core-9.0.0-20241026.042336.jar:9.0.0-20241026.042336]
	at org.frankframework.receivers.Receiver.processMessageInAdapter(Receiver.java:1258) ~[frankframework-core-9.0.0-20241026.042336.jar:9.0.0-20241026.042336]
	at org.frankframework.receivers.Receiver.processRequest(Receiver.java:1051) ~[frankframework-core-9.0.0-20241026.042336.jar:9.0.0-20241026.042336]
	at org.frankframework.http.PushingListenerAdapter.processRequest(PushingListenerAdapter.java:104) ~[frankframework-core-9.0.0-20241026.042336.jar:9.0.0-20241026.042336]
	at org.frankframework.http.WebServiceListener.processRequest(WebServiceListener.java:211) ~[frankframework-core-9.0.0-20241026.042336.jar:9.0.0-20241026.042336]
	at org.frankframework.http.cxf.MessageProvider.processRequest(MessageProvider.java:47) ~[frankframework-core-9.0.0-20241026.042336.jar:9.0.0-20241026.042336]
	at org.frankframework.http.cxf.SOAPProviderBase.invoke(SOAPProviderBase.java:164) ~[frankframework-core-9.0.0-20241026.042336.jar:9.0.0-20241026.042336]
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(Unknown Source) ~[?:?]
	... 
	at org.apache.cxf.service.invoker.AbstractInvoker.performInvocation(AbstractInvoker.java:179) ~[cxf-core-4.0.5.jar:4.0.5]
	at org.apache.cxf.jaxws.JAXWSMethodInvoker.performInvocation(JAXWSMethodInvoker.java:65) ~[cxf-rt-frontend-jaxws-4.0.5.jar:4.0.5]
	at org.apache.cxf.service.invoker.AbstractInvoker.invoke(AbstractInvoker.java:96) ~[cxf-core-4.0.5.jar:4.0.5]
	at org.apache.cxf.jaxws.AbstractJAXWSMethodInvoker.invoke(AbstractJAXWSMethodInvoker.java:231) ~[cxf-rt-frontend-jaxws-4.0.5.jar:4.0.5]
	at org.apache.cxf.jaxws.JAXWSMethodInvoker.invoke(JAXWSMethodInvoker.java:84) ~[cxf-rt-frontend-jaxws-4.0.5.jar:4.0.5]
	at org.apache.cxf.service.invoker.AbstractInvoker.invoke(AbstractInvoker.java:74) ~[cxf-core-4.0.5.jar:4.0.5]
	at org.apache.cxf.interceptor.ServiceInvokerInterceptor$1.run(ServiceInvokerInterceptor.java:59) ~[cxf-core-4.0.5.jar:4.0.5]
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) ~[?:?]
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source) ~[?:?]
	at org.apache.cxf.interceptor.ServiceInvokerInterceptor$2.run(ServiceInvokerInterceptor.java:126) ~[cxf-core-4.0.5.jar:4.0.5]
	at org.apache.cxf.workqueue.SynchronousExecutor.execute(SynchronousExecutor.java:37) ~[cxf-core-4.0.5.jar:4.0.5]
	at org.apache.cxf.interceptor.ServiceInvokerInterceptor.handleMessage(ServiceInvokerInterceptor.java:131) ~[cxf-core-4.0.5.jar:4.0.5]
	at org.apache.cxf.phase.PhaseInterceptorChain.doIntercept(PhaseInterceptorChain.java:307) ~[cxf-core-4.0.5.jar:4.0.5]
	at org.apache.cxf.transport.ChainInitiationObserver.onMessage(ChainInitiationObserver.java:121) ~[cxf-core-4.0.5.jar:4.0.5]
	at org.apache.cxf.transport.http.AbstractHTTPDestination.invoke(AbstractHTTPDestination.java:265) ~[cxf-rt-transports-http-4.0.5.jar:4.0.5]
	at org.apache.cxf.transport.servlet.ServletController.invokeDestination(ServletController.java:233) ~[cxf-rt-transports-http-4.0.5.jar:4.0.5]
	at org.apache.cxf.transport.servlet.ServletController.invoke(ServletController.java:207) ~[cxf-rt-transports-http-4.0.5.jar:4.0.5]
	at org.apache.cxf.transport.servlet.ServletController.invoke(ServletController.java:159) ~[cxf-rt-transports-http-4.0.5.jar:4.0.5]
	at org.apache.cxf.transport.servlet.CXFNonSpringServlet.invoke(CXFNonSpringServlet.java:224) ~[cxf-rt-transports-http-4.0.5.jar:4.0.5]
	at org.apache.cxf.transport.servlet.AbstractHTTPServlet.handleRequest(AbstractHTTPServlet.java:303) ~[cxf-rt-transports-http-4.0.5.jar:4.0.5]
	at org.apache.cxf.transport.servlet.AbstractHTTPServlet.doPost(AbstractHTTPServlet.java:216) ~[cxf-rt-transports-http-4.0.5.jar:4.0.5]
	at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:590) ~[servlet-api.jar:6.0]
	at org.apache.cxf.transport.servlet.AbstractHTTPServlet.service(AbstractHTTPServlet.java:278) ~[cxf-rt-transports-http-4.0.5.jar:4.0.5]
	at org.frankframework.http.cxf.SoapProviderServlet.service(SoapProviderServlet.java:44) ~[frankframework-core-9.0.0-20241026.042336.jar:9.0.0-20241026.042336]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:195) ~[catalina.jar:10.1.28]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140) ~[catalina.jar:10.1.28]
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:51) ~[tomcat-websocket.jar:10.1.28]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164) ~[catalina.jar:10.1.28]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140) ~[catalina.jar:10.1.28]
	... suppressed 18 lines
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116) ~[spring-web-6.1.14.jar:6.1.14]
	... suppressed 5 lines
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116) ~[spring-web-6.1.14.jar:6.1.14]
	... suppressed 2 lines
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116) ~[spring-web-6.1.14.jar:6.1.14]
	... suppressed 3 lines
	at org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:362) ~[spring-web-6.1.14.jar:6.1.14]
	at org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:278) ~[spring-web-6.1.14.jar:6.1.14]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164) ~[catalina.jar:10.1.28]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140) ~[catalina.jar:10.1.28]
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:167) ~[catalina.jar:10.1.28]
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:90) ~[catalina.jar:10.1.28]
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:598) ~[catalina.jar:10.1.28]
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:115) ~[catalina.jar:10.1.28]
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:93) ~[catalina.jar:10.1.28]
	at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:663) ~[catalina.jar:10.1.28]
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74) ~[catalina.jar:10.1.28]
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:344) ~[catalina.jar:10.1.28]
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:384) ~[tomcat-coyote.jar:10.1.28]
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63) ~[tomcat-coyote.jar:10.1.28]
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:904) ~[tomcat-coyote.jar:10.1.28]
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1741) ~[tomcat-coyote.jar:10.1.28]
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52) ~[tomcat-coyote.jar:10.1.28]
	at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1190) ~[tomcat-util.jar:10.1.28]
	at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659) ~[tomcat-util.jar:10.1.28]
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:63) ~[tomcat-util.jar:10.1.28]
	at java.base/java.lang.Thread.run(Unknown Source) [?:?]
Caused by: org.frankframework.core.SenderException: Cannot transform input: (XPathException) I/O error reported by XML parser processing null: (IOException) Stream closed
	at org.frankframework.senders.XsltSender.sendMessage(XsltSender.java:314) ~[frankframework-core-9.0.0-20241026.042336.jar:9.0.0-20241026.042336]
	at org.frankframework.pipes.XsltPipe.doPipe(XsltPipe.java:109) ~[frankframework-core-9.0.0-20241026.042336.jar:9.0.0-20241026.042336]
	... 182 more
Caused by: net.sf.saxon.trans.XPathException: I/O error reported by XML parser processing null: Stream closed
	at net.sf.saxon.event.Sender.sendSAXSource(Sender.java:467) ~[Saxon-HE-10.9.jar:?]
	at net.sf.saxon.event.Sender.send(Sender.java:168) ~[Saxon-HE-10.9.jar:?]
	at net.sf.saxon.Controller.makeSourceTree(Controller.java:1320) ~[Saxon-HE-10.9.jar:?]
	at net.sf.saxon.s9api.XsltTransformer.transform(XsltTransformer.java:335) ~[Saxon-HE-10.9.jar:?]
	at net.sf.saxon.jaxp.TransformerImpl.transform(TransformerImpl.java:74) ~[Saxon-HE-10.9.jar:?]
	at org.frankframework.util.XmlUtils.transformXml(XmlUtils.java:1355) ~[frankframework-core-9.0.0-20241026.042336.jar:?]
	at org.frankframework.util.XmlUtils.transformXml(XmlUtils.java:1345) ~[frankframework-core-9.0.0-20241026.042336.jar:?]
	at org.frankframework.util.TransformerPool.transform(TransformerPool.java:479) ~[frankframework-core-9.0.0-20241026.042336.jar:?]
	at org.frankframework.util.TransformerPool.transform(TransformerPool.java:427) ~[frankframework-core-9.0.0-20241026.042336.jar:?]
	at org.frankframework.util.TransformerPool.transform(TransformerPool.java:423) ~[frankframework-core-9.0.0-20241026.042336.jar:?]
	at org.frankframework.util.XmlUtils.getXsltConfig(XmlUtils.java:237) ~[frankframework-core-9.0.0-20241026.042336.jar:?]
	at org.frankframework.util.TransformerPool.getConfigMap(TransformerPool.java:530) ~[frankframework-core-9.0.0-20241026.042336.jar:?]
	at org.frankframework.util.TransformerPool.getDisableOutputEscaping(TransformerPool.java:560) ~[frankframework-core-9.0.0-20241026.042336.jar:?]
	at org.frankframework.senders.XsltSender.isDisableOutputEscaping(XsltSender.java:175) ~[frankframework-core-9.0.0-20241026.042336.jar:9.0.0-20241026.042336]
	at org.frankframework.senders.XsltSender.createHandler(XsltSender.java:229) ~[frankframework-core-9.0.0-20241026.042336.jar:9.0.0-20241026.042336]
	at org.frankframework.senders.XsltSender.sendMessage(XsltSender.java:299) ~[frankframework-core-9.0.0-20241026.042336.jar:9.0.0-20241026.042336]
	at org.frankframework.pipes.XsltPipe.doPipe(XsltPipe.java:109) ~[frankframework-core-9.0.0-20241026.042336.jar:9.0.0-20241026.042336]
	... 182 more
Caused by: java.io.IOException: Stream closed
	at java.base/java.io.StringReader.ensureOpen(Unknown Source) ~[?:?]
	at java.base/java.io.StringReader.read(Unknown Source) ~[?:?]
	at org.apache.xerces.impl.XMLEntityScanner.load(Unknown Source) ~[ibis-xerces-2.12.2-xml-schema-1.1.jar:2.12.2-xml-schema-1.1]
	at org.apache.xerces.impl.XMLEntityScanner.skipString(Unknown Source) ~[ibis-xerces-2.12.2-xml-schema-1.1.jar:2.12.2-xml-schema-1.1]
	at org.apache.xerces.impl.XMLVersionDetector.determineDocVersion(Unknown Source) ~[ibis-xerces-2.12.2-xml-schema-1.1.jar:2.12.2-xml-schema-1.1]
	at org.apache.xerces.parsers.XML11Configuration.parse(Unknown Source) ~[ibis-xerces-2.12.2-xml-schema-1.1.jar:?]
	at org.apache.xerces.parsers.XML11Configuration.parse(Unknown Source) ~[ibis-xerces-2.12.2-xml-schema-1.1.jar:?]
	at org.apache.xerces.parsers.XMLParser.parse(Unknown Source) ~[ibis-xerces-2.12.2-xml-schema-1.1.jar:?]
	at org.apache.xerces.parsers.AbstractSAXParser.parse(Unknown Source) ~[ibis-xerces-2.12.2-xml-schema-1.1.jar:?]
	at org.apache.xerces.jaxp.SAXParserImpl$JAXPSAXParser.parse(Unknown Source) ~[ibis-xerces-2.12.2-xml-schema-1.1.jar:?]
	at net.sf.saxon.event.Sender.sendSAXSource(Sender.java:439) ~[Saxon-HE-10.9.jar:?]
	at net.sf.saxon.event.Sender.send(Sender.java:168) ~[Saxon-HE-10.9.jar:?]
	at net.sf.saxon.Controller.makeSourceTree(Controller.java:1320) ~[Saxon-HE-10.9.jar:?]
	at net.sf.saxon.s9api.XsltTransformer.transform(XsltTransformer.java:335) ~[Saxon-HE-10.9.jar:?]
	at net.sf.saxon.jaxp.TransformerImpl.transform(TransformerImpl.java:74) ~[Saxon-HE-10.9.jar:?]
	at org.frankframework.util.XmlUtils.transformXml(XmlUtils.java:1355) ~[frankframework-core-9.0.0-20241026.042336.jar:?]
	at org.frankframework.util.XmlUtils.transformXml(XmlUtils.java:1345) ~[frankframework-core-9.0.0-20241026.042336.jar:?]
	at org.frankframework.util.TransformerPool.transform(TransformerPool.java:479) ~[frankframework-core-9.0.0-20241026.042336.jar:?]
	at org.frankframework.util.TransformerPool.transform(TransformerPool.java:427) ~[frankframework-core-9.0.0-20241026.042336.jar:?]
	at org.frankframework.util.TransformerPool.transform(TransformerPool.java:423) ~[frankframework-core-9.0.0-20241026.042336.jar:?]
	at org.frankframework.util.XmlUtils.getXsltConfig(XmlUtils.java:237) ~[frankframework-core-9.0.0-20241026.042336.jar:?]
	at org.frankframework.util.TransformerPool.getConfigMap(TransformerPool.java:530) ~[frankframework-core-9.0.0-20241026.042336.jar:?]
	at org.frankframework.util.TransformerPool.getDisableOutputEscaping(TransformerPool.java:560) ~[frankframework-core-9.0.0-20241026.042336.jar:?]
	at org.frankframework.senders.XsltSender.isDisableOutputEscaping(XsltSender.java:175) ~[frankframework-core-9.0.0-20241026.042336.jar:9.0.0-20241026.042336]
	at org.frankframework.senders.XsltSender.createHandler(XsltSender.java:229) ~[frankframework-core-9.0.0-20241026.042336.jar:9.0.0-20241026.042336]
	at org.frankframework.senders.XsltSender.sendMessage(XsltSender.java:299) ~[frankframework-core-9.0.0-20241026.042336.jar:9.0.0-20241026.042336]
	at org.frankframework.pipes.XsltPipe.doPipe(XsltPipe.java:109) ~[frankframework-core-9.0.0-20241026.042336.jar:9.0.0-20241026.042336]
	... 182 more
2025-01-09 09:47:41,904 INFO  [http-nio-8080-exec-10] {mid=ac150003--773cd1c0_1944a3edfa2_-7fd0, cid=ac150003--773cd1c0_1944a3edfa2_-7fd0, adapter=SoapEndpointRouter_VrijeBerichten, pipe=WrapFo03Response} util.XmlUtils - omitting setting of parameter [errorCode] on transformer, as it has a null-value
2025-01-09 09:47:41,904 INFO  [http-nio-8080-exec-10] {mid=ac150003--773cd1c0_1944a3edfa2_-7fd0, cid=ac150003--773cd1c0_1944a3edfa2_-7fd0, adapter=SoapEndpointRouter_VrijeBerichten, pipe=WrapFo03Response} util.XmlUtils - omitting setting of parameter [errorReason] on transformer, as it has a null-value
2025-01-09 09:47:41,904 INFO  [http-nio-8080-exec-10] {mid=ac150003--773cd1c0_1944a3edfa2_-7fd0, cid=ac150003--773cd1c0_1944a3edfa2_-7fd0, adapter=SoapEndpointRouter_VrijeBerichten, pipe=WrapFo03Response} util.XmlUtils - omitting setting of parameter [errorDetailText] on transformer, as it has a null-value
2025-01-09 09:47:41,912 INFO  [http-nio-8080-exec-10] {mid=ac150003--773cd1c0_1944a3edfa2_-7fd0, cid=ac150003--773cd1c0_1944a3edfa2_-7fd0, adapter=SoapEndpointRouter_VrijeBerichten} core.Adapter - Adapter [SoapEndpointRouter_VrijeBerichten] Pipeline finished processing message with messageId [ac150003--773cd1c0_1944a3edfa2_-7fd0] with exit-state [SUCCESS]
```


### Anything else?

_No response_

--- a/core/src/main/java/org/frankframework/util/TransformerPool.java
+++ b/core/src/main/java/org/frankframework/util/TransformerPool.java
@@ -526,16 +526,17 @@ public class TransformerPool {
 		return new TransformerFilter(threadConnector, getTransformerHandler(), handler, removeNamespacesFromInput, handleLexicalEvents);
 	}
 
-	@Nonnull
-	public Map<String, String> getConfigMap() throws TransformerException, IOException {
-		if (configMap == null) {
-			configMap = XmlUtils.getXsltConfig(configSource);
+
+	public @Nonnull Map<String, String> getConfigMap() throws TransformerException, IOException {
+		synchronized (configSource) {
+			if (configMap == null) {
+				configMap = XmlUtils.getXsltConfig(configSource);
+			}
 		}
 		return configMap;
 	}
 
-	@Nullable
-	public Boolean getOmitXmlDeclaration() throws TransformerException, IOException {
+	public @Nullable Boolean getOmitXmlDeclaration() throws TransformerException, IOException {
 		String setting = getConfigMap().get("output-omit-xml-declaration");
 		if (setting == null) {
 			return null;
@@ -543,8 +544,7 @@ public class TransformerPool {
 		return "yes".equals(setting);
 	}
 
-	@Nullable
-	public Boolean getIndent() throws TransformerException, IOException {
+	public @Nullable Boolean getIndent() throws TransformerException, IOException {
 		String setting = getConfigMap().get("output-indent");
 		if (setting == null) {
 			return null;
@@ -552,13 +552,11 @@ public class TransformerPool {
 		return "yes".equals(setting);
 	}
 
-	@Nullable
-	public String getOutputMethod() throws TransformerException, IOException {
+	public @Nullable String getOutputMethod() throws TransformerException, IOException {
 		return getConfigMap().get("output-method");
 	}
 
-	@Nullable
-	public Boolean getDisableOutputEscaping() throws TransformerException, IOException {
+	public @Nullable Boolean getDisableOutputEscaping() throws TransformerException, IOException {
 		String setting = getConfigMap().get("disable-output-escaping");
 		if (setting == null) {
 			return null;

--- a/core/src/main/java/org/frankframework/util/TransformerPool.java
+++ b/core/src/main/java/org/frankframework/util/TransformerPool.java
@@ -528,6 +528,7 @@ public class TransformerPool {
 
 
 	public @Nonnull Map<String, String> getConfigMap() throws TransformerException, IOException {
+		// Due to lazy-loading of the config-map this can happen in multiple threads simultaneously. Hence we synchronize here, the config-source seems a logical choice that doesn't lock too much.
 		synchronized (configSource) {
 			if (configMap == null) {
 				configMap = XmlUtils.getXsltConfig(configSource);

